sdk/src/Services/DynamoDBv2/Custom/DataModel/IDynamoDBContextBuilder.cs

```diff
@@ -0,0 +1,49 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System;
+
+namespace Amazon.DynamoDBv2.DataModel
+{
+    /// <summary>
+    /// Interface for a builder that constructs a <see cref="DynamoDBContext"/>
+    /// </summary>
+    public interface IDynamoDBContextBuilder : IDisposable
+    {
+        /// <summary>
+        /// Sets the DynamoDB client to be used by the <see cref="DynamoDBContext"/> that is constructed
```

Comment:
Add a comment that if a client is not provided then a client will be created for users using the environmental to search for credentials and region configuration.

Reply:
Removed the 2 set methods, but I added this comment to the new `WithDynamoDBClient`